src/app/models/workspace.server.model.ts

```diff
@@ -53,6 +53,19 @@ const compileWorkspaceModel = (db: Mongoose): IWorkspaceModel => {
     return this.create({ title, admin, formIds: [] })
   }
 
+  WorkspaceSchema.statics.updateWorkspaceTitle = async function ({
+    title,
+    workspaceId,
+    admin,
+  }: {
+    title: string
+    workspaceId: IWorkspaceSchema['_id']
+    admin: IUserSchema['_id']
+  }) {
+    await this.updateOne({ _id: workspaceId, admin }, { title: title }).exec()
+    return this.findOne({ _id: workspaceId, admin }).exec()
```

Comment:
rather than using 2 db calls, just use `findOneAndUpdate` with the `new` argument which makes `mongoose` return the new document.

in a situation where you think a library is doing something unintuitive (like forcing you to make 2 db calls to update a document and get back the updated version), there's probably a better way to do things!

Reply:
also why do we need to pass the `admin`? the `workspaceId` should be sufficient right?

Reply:
Okay will update to use `findOneAndUpdate`! 

Yeah `workspaceId` is sufficient. I thought is just "defensive" so that users can't update workspace that don't belong to them (though this check is already done in the controller)

Reply:
I'll remove the `admin` from the filter query!

Reply:
hmm okay. the way I think about this is that by the time the query reaches the model layer, those business logic checks should already have been done. but then again we have such checks in the model validation layer...up to you!

Reply:
> the way I think about this is that by the time the query reaches the model layer, those business logic checks should already have been done

Yep agreed, I've removed `admin` check from the model!